scargo/file_generators/templates/CMakeLists.txt.j2

```diff
@@ -6,11 +6,7 @@
 # DO NOT EDIT THIS FILE!
 # This file is generated by `scargo update`.
 #
-{% if config.project.target.family == "esp32" %}
-cmake_minimum_required(VERSION 3.14)
-{% else %}
-cmake_minimum_required(VERSION 3.16)
-{% endif %}
+cmake_minimum_required(VERSION 3.22)
```

Comment:
Yes, I'm sure it works. This should be summarized as follows:
1. Conan cmake (CMakeDeps) generators require new cmake to work nicely (3.18 as far as I remember)—it's a reason for change.
2. Deafult cmake for Ubuntu 22.04 wchich we shoud use today is 3.22 - reason why i used this particular wersion
3. I tested it with the default esp32 docker images, and it worked (we had version 3.25 already installed inside docker).
4. The old implementation is "Because of reasons," so no one writes why it has a condition.
5. Old implementation use version 3.14 which is something prehistoric and i do not see point in maintaing it?
6. As rule of thumb cmake minimal requirements update is rather stable change.